nucypher/blockchain/eth/utils.py

```diff
@@ -162,78 +160,22 @@ def get_default_rpc_endpoints() -> Dict[int, List[str]]:
         return {}
 
 
-def get_healthy_default_rpc_endpoints(chain_id: int) -> List[str]:
+def get_healthy_default_rpc_endpoints(domain: str) -> Dict[int, List[str]]:
     """Returns a list of healthy RPC endpoints for a given chain ID."""
 
-    endpoints = get_default_rpc_endpoints()
-    chain_endpoints = endpoints.get(chain_id)
-
-    if not chain_endpoints:
-        LOGGER.error(f"No default RPC endpoints found for chain ID {chain_id}")
-        return list()
-
-    healthy = [
-        endpoint for endpoint in chain_endpoints if rpc_endpoint_health_check(endpoint)
-    ]
-    LOGGER.info(f"Healthy default RPC endpoints for chain ID {chain_id}: {healthy}")
-    if not healthy:
-        LOGGER.warn(
-            f"No healthy default RPC endpoints available for chain ID {chain_id}"
-        )
+    endpoints = get_default_rpc_endpoints(domain)
+
+    if domain == "mainnet":
+        # iterate over all chains and filter out unhealthy endpoints
+        healthy = {
+            chain_id: [
+                endpoint
+                for endpoint in endpoints[chain_id]
+                if rpc_endpoint_health_check(endpoint)
+            ]
+            for chain_id in endpoints
```

Comment:
We should randomize the list of endpoints at some point.  The reason being to prevent the first URL in the list from potentially being used by default by all nodes.

Options include:
a) here - i.e. when it is obtained from he remote json file
b) when we are iterating over the endpoints for conditions. That being said, this is tougher since we should bias for user supplied endpoints. Maybe a) would be better?

Thoughts?

Reply:
Yeah i also think a)

I'll sort that out now